### PR TITLE
Edit hack lambda

### PR DESCRIPTION
**Pull request by HuangStanley050**
Create end point for editing a hack, at the moment still no protection set in place for our lambda end points, however at standup with Thomas and Joseph, we agreed to have protection set for the POST routes and just leave the GETS routes /hacks /hack-detail for non authenticated user.

Options on the table are something called AWS event gateway or AWS authorizer? Waiting for the card to be created and also we have removed the end point to return user emails.

### CODE REVIEW

File: editHack.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    });
+    conn.model(
+      "Hack",
+      new mongoose.Schema({ title: String, description: String, goal: String })
+    );
+  }
+  const Hack = conn.model("Hack");
+
+  // guard against empty fields
+  if (goal) update.goal = goal;
```
Comment (jpeyper) [inline editHack.js:30]: so `undefined`, `null` or `""` will all be missing from the document? these are not guarded against in the addHack lambda (https://github.com/hackthehack/hack-zero-v2-backend/blob/master/addHack.js)

File: editHack.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  if (title) update.title = title;
+  if (description) update.description = description;
+
+  try {
+    const result = await Hack.findByIdAndUpdate(hackId, update, { new: true });
```
Comment (jpeyper) [inline editHack.js]: if you did an upsert, would you be able to use the same lambda for adding and editing a hack?

**Review by jpeyper — APPROVED** [review]

File: editHack.js (hunk 1 above)
Comment (HuangStanley050) [reply editHack.js:30]: Valid point. Might include a default value when document is created in mongodb. Will address that in addHack.js before merge to master👍

File: editHack.js (hunk 2 above)
Comment (HuangStanley050) [reply editHack.js]: Can research upsert, originally wanted to reuse addHack but decided to keep it lean

Comment (HuangStanley050) [reply editHack.js]: Upsert looks like a good option with minimum logic involved. Will test it out tomorrow

Comment (josephgarner) [inline editHack.js]: At the moment the returned team is only populated with team members ID's, is it possible to attach populate to ` const result = await Hack.findByIdAndUpdate(hackId, update, { new: true });` so we can also grab the members names?

Comment (HuangStanley050) [reply editHack.js]: yes definitely and also just wondering if need to somehow guard against the people editing other people's hack or is that already implemented in the front end?

**Pull request merged**